src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -130,6 +130,9 @@ static NamesAndTypesList getColumnsList(const ASTExpressionList * columns_defini
                     child = new_child;
                 }
             }
+
+            if (type_name_upper == "DATE")
+                data_type_function->name = "Date32";
```

Comment:
Need to pass setting here. Date32 is not stable, so better not to use it regardless. How did it work before without these lines?

Reply:
for now MySQLDataTypesSupport just work for Mysql database or table or function, not MaterializedMySQL(The option to synchronize different data types is not supported for now ), if need supported, We have some other works to do.  like https://github.com/ClickHouse/ClickHouse/issues/31469 I said，It can try to meet most of our needs by use date32 in MaterializedMySQL.